adal/src/androidTest/java/com/microsoft/aad/adal/OauthTests.java

```diff
@@ -276,6 +276,16 @@ public void testGetCodeRequestUrlWithClaimsInExtraQP() throws UnsupportedEncodin
         assertTrue(codeRequestUrl.contains("claims=testclaims&a=b"));
     }
 
+    public void testGetCodeRequestUrlWithInstanceAwareInExtraQP() throws UnsupportedEncodingException {
```

Comment:
you should also have a test to make sure that you're constructing the request to token endpoint correctly, i.e. if cloud instance name is returned correctly and tenant name is provided is provided in the original request vs common is used.

Reply:
Good point! added the test to testprocessUIResponseParamsForSovereignCloud